roles/docker/tasks/main.yml

```diff
@@ -24,5 +24,15 @@
   shell: systemctl daemon-reload && systemctl stop docker && systemctl start docker-tcp.socket && systemctl start docker
   when: docker_env_conf_result.changed or docker_tcp_conf_result.changed or docker_svc_conf_result.changed
 
+#NOTE: Without this step, pip crashes after installing docker-py/docker-compose
+#      Move to appropriate role later
+- name: upgrade pip
```

Comment:
Upgrading pip is necessary here as docker-compose module which is done after this task is also crashing pip... the issue is not visible immediately, but when pip module is used later, it shows up. It might be difficult to debug this issue later without much context.